sql/core/src/main/scala/org/apache/spark/sql/streaming/progress.scala

```diff
@@ -163,7 +163,8 @@ class SourceProgress protected[sql](
   val endOffset: String,
   val numInputRows: Long,
   val inputRowsPerSecond: Double,
-  val processedRowsPerSecond: Double) extends Serializable {
+  val processedRowsPerSecond: Double,
+  val customMetrics: Option[JValue] = None) extends Serializable {
```

Comment:
Wait .. this is an exposed API, right? I guess this is exposed to Java API too (for instance `query.lastProgress().sources()`)? In that case, we should avoid Scala's Option and `org.json4s.*`. If this is supposed to be hidden here, I think we should better find a way to hide this with package level access modifier.

Reply:
@HyukjinKwon 
Nice finding. I missed it while reviewing.

Btw, FYI, in #21469 I'm adding new field with default value in StateOperatorProgress, like `val customMetrics: ju.Map[String, JLong] = new ju.HashMap()` and MiMa doesn't complain.

https://github.com/apache/spark/pull/21469/files#diff-e09301244e3c6b1a69eda6c4bd2ddb52

@arunmahadevan 
Maybe `ju.Map[String, JLong]` will also work here.

Reply:
Refactored to Json String instead of JValue.